cli/azd/pkg/environment/manager.go

```diff
@@ -310,6 +306,15 @@ func (m *manager) Get(ctx context.Context, name string) (*Environment, error) {
 		localEnv = remoteEnv
 	}
 
+	// Ensures local environment variable name is synced with the environment name
+	envName, ok := localEnv.LookupEnv(EnvNameEnvVarName)
+	if !ok || envName != name {
+		localEnv.DotenvSet(EnvNameEnvVarName, name)
+		if err := m.Save(ctx, localEnv); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
This is the minor code change.

Reply:
Why not self recover when we can for a simple use case like this?  If subscription id or location was removed the provision provider would recover and prompt the users for values again.  

In this case we already know what the environment name is - just ensuring it is persisted in the .env file to support other downstream scenarios.